lib/dao.js

```diff
@@ -1562,7 +1565,23 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
     var DataType = props[p] && props[p].type;
+    if (p.match(/\./)) {
```

Comment:
Please move this code above `DataType`, since `DataType` is not used for nested properties.